src/xrpld/app/tx/detail/LoanSet.cpp

```diff
@@ -0,0 +1,675 @@
+            return ter;
+    }
+
+    if (auto const ter = accountSendMulti(
+            view,
+            vaultPseudo,
+            vaultAsset,
+            {{borrower, loanAssetsToBorrower}, {brokerOwner, originationFee}},
+            j_,
+            WaiveTransferFee::Yes))
+        return ter;
+
+    // Get shortcuts to the loan property values
+    auto const startDate = getStartDate(view);
+    auto loanSequenceProxy = brokerSle->at(sfLoanSequence);
+
+    // Create the loan
+    auto loan =
+        std::make_shared<SLE>(keylet::loan(brokerID, *loanSequenceProxy));
+
+    // Prevent copy/paste errors
+    auto setLoanField =
+        [&loan, &tx](auto const& field, std::uint32_t const defValue = 0) {
+            // at() is smart enough to unseat a default field set to the default
+            // value
+            loan->at(field) = tx[field].value_or(defValue);
+        };
+
+    // Set required and fixed tx fields
+    loan->at(sfLoanScale) = properties.loanScale;
+    loan->at(sfStartDate) = startDate;
+    loan->at(sfPaymentInterval) = paymentInterval;
+    loan->at(sfLoanSequence) = *loanSequenceProxy;
+    loan->at(sfLoanBrokerID) = brokerID;
+    loan->at(sfBorrower) = borrower;
+    // Set all other transaction fields directly from the transaction
+    if (tx.isFlag(tfLoanOverpayment))
+        loan->setFlag(lsfLoanOverpayment);
+    setLoanField(~sfLoanOriginationFee);
+    setLoanField(~sfLoanServiceFee);
+    setLoanField(~sfLatePaymentFee);
+    setLoanField(~sfClosePaymentFee);
+    setLoanField(~sfOverpaymentFee);
+    setLoanField(~sfInterestRate);
+    setLoanField(~sfLateInterestRate);
+    setLoanField(~sfCloseInterestRate);
+    setLoanField(~sfOverpaymentInterestRate);
+    setLoanField(~sfGracePeriod, defaultGracePeriod);
+    // Set dynamic / computed fields to their initial values
+    loan->at(sfPrincipalOutstanding) = principalRequested;
+    loan->at(sfPeriodicPayment) = properties.periodicPayment;
+    loan->at(sfTotalValueOutstanding) = properties.totalValueOutstanding;
+    loan->at(sfManagementFeeOutstanding) = properties.managementFeeOwedToBroker;
+    loan->at(sfPreviousPaymentDate) = 0;
+    loan->at(sfNextPaymentDueDate) = startDate + paymentInterval;
+    loan->at(sfPaymentRemaining) = paymentTotal;
+    view.insert(loan);
+
+    // Update the balances in the vault
+    vaultAvailableProxy -= principalRequested;
+    vaultTotalProxy += state.interestDue;
+    XRPL_ASSERT_PARTS(
+        *vaultAvailableProxy <= *vaultTotalProxy,
+        "ripple::LoanSet::doApply",
+        "assets available must not be greater than assets outstanding");
+    view.update(vaultSle);
+
+    // Update the balances in the loan broker
+    brokerSle->at(sfDebtTotal) += newDebtDelta;
+    // The broker's owner count is solely for the number of outstanding loans,
+    // and is distinct from the broker's pseudo-account's owner count
+    adjustOwnerCount(view, brokerSle, 1, j_);
+    loanSequenceProxy += 1;
```

Comment:
although unlikely, what if loan sequence is at the max uint32? is it intentional to wrap it to 0?

Reply:
> although unlikely, what if loan sequence is at the max uint32? is it intentional to wrap it to 0?

Good catch! It absolutely should not roll over. Will fix.